libraries/chain/block_state.cpp

```diff
@@ -337,11 +337,23 @@ finality_data_t block_state::get_finality_data() {
    if (!base_digest) {
       base_digest = compute_base_digest(); // cache it
    }
+
+   // Check if there is any proposed finalizer policy in the block
+   std::optional<finalizer_policy> proposed_finalizer_policy = std::nullopt;
+   auto range = finalizer_policies.equal_range(block_num());
+   for (auto itr = range.first; itr != range.second; ++itr) {
+      if (itr->second.state == finalizer_policy_tracker::state_t::proposed) {
+         proposed_finalizer_policy = *itr->second.policy;
+         break;
+      }
+   }
+
    return {
       // other fields take the default values set by finality_data_t definition
       .active_finalizer_policy_generation = active_finalizer_policy->generation,
-      .action_mroot = action_mroot,
-      .base_digest  = *base_digest
+      .action_mroot                       = action_mroot,
+      .base_digest                        = *base_digest,
+      .proposed_finalizer_policy          = std::move(proposed_finalizer_policy)
    };
```

Comment:
I don't think this does the correct thing for IF Genesis Block. The IF Genesis Block has a finalizer set proposed in it (that is what makes it the IF Genesis Block), but that finalizer policy is not put in the tracker I don't think. Is that an issue or is IF Genesis Block handled specially for this finality data?

Reply:
Should `create_if_genesis_block` set the first proposed policy in `finalizer_policies`?

Reply:
It's decided at the standup meeting we will treat this as a special case here and return the active policy for Savanna Genesis block. Thanks @heifner